src/main/java/ProjectOcean/Controller/StudyPlansController.java

```diff
@@ -0,0 +1,116 @@
+import javafx.scene.control.Alert;
+import javafx.scene.control.Button;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+import java.io.IOException;
+
+/**
+ * Represents a graphical component of a study plan.
+ */
+public class StudyPlansController extends AnchorPane {
+
+    @FXML private VBox studyPlanContainer;
+    @FXML private Button addButton;
+
+    private CoursePlanningSystem model;
+    private ShowAStudyPlan showStudyPlan;
+
+    public StudyPlansController(CoursePlanningSystem model, ShowAStudyPlan showStudyPlan) {
+
+        this.model = model;
+        this.showStudyPlan = showStudyPlan;
+
+        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource(
+                "/StudyPlansWindow.fxml"));
+        fxmlLoader.setRoot(this);
+        fxmlLoader.setController(this);
+
+        try {
+            fxmlLoader.load();
+        } catch (IOException exception) {
+            throw new RuntimeException(exception);
+        }
+
+        displayAllStudyPlans();
+
+    }
+
+    @FXML
+    private void onAddStudyPlanClicked(MouseEvent event) {
+
+        deactivateCurrStudyPlanButton();
+        model.addStudyPlan(); // Method also setting new studyPlan to current.
+
+        displayAllStudyPlans();
+        showStudyPlan.showAStudyPlan();
+
+        event.consume();
+
+    }
+
+    private void displayAllStudyPlans() {
+        studyPlanContainer.getChildren().clear();
+        studyPlanContainer.getChildren().add(addButton);
+
+        for (StudyPlan studyPlan : model.getStudent().getAllStudyPlans()) {
+            // Set new buttons properties, values taken from button created in ApplicationWindow.fxml
+            int nOfStudyPlans = studyPlanContainer.getChildren().size();
+            ButtonController newButton = new ButtonController(model, showStudyPlan, this::deactivateCurrStudyPlanButton, nOfStudyPlans, studyPlan);
+
+            studyPlanContainer.getChildren().add(nOfStudyPlans - 1, newButton);
+
+            if (newButton.getStudyPlan() == model.getStudent().getCurrentStudyPlan()) {
+                newButton.activateDefaultButton();
+            }
+        }
+    }
+
+    private void deactivateCurrStudyPlanButton() {
+        ButtonController buttonController = getCurrStudyPlansButtonController();
+        if (buttonController != null) {
+            buttonController.deActivateDefaultButton();
+        }
+    }
+
+    // Premise: there is an active study plan button.
+    private ButtonController getCurrStudyPlansButtonController() {
+        ButtonController bc = null;
+        for (int i = 0; i < model.getStudent().getAllStudyPlans().size(); i++) {
+            ButtonController buttonController = (ButtonController) studyPlanContainer.getChildren().get(i);
+            if (buttonController.getStudyPlan() == model.getStudent().getCurrentStudyPlan()) {
+                bc = buttonController;
+            }
+        }
+        return bc;
+    }
```

Comment:
What if buttonController is never set? Does it need to be null, can you not just declare it? 
Also, why do you take size of mode study plan list instead of get children size? That is what your actually goingt through?

You could probably do a for(Child c : studyPlanContainer.getChildren) or similar instead and it would look better.

Reply:
It's true. The problem occurring from having both Button and StudyPlanButtonController objects in same VBox. Iterating the VBox and cast the child Node to StudyPlanButtonController throws an error (failing when cast Button to StudyPlanButtonController).
The Button is the '+'-button, which is placed inside the VBox.
A fix would be to change the FXML and place the Button outside of the VBox. This would actually make some loops simpler (using a foreach-loop) and this method to not return a null in the worst case.
Should this change be done?

Reply:
Well, it might be possible but scrollPane in Scene builder/JavaFX is not my friend atm. Could need som help with that.